Adding a comment model and admin model registration

maayanhd opened the pull request:
- opening a separate comment_system app to maximize parallelism on
  models
- adding [migrations/0001_initial.py][1] and a model ([models.py][2]) for representing a comment.
- adding customized model registration to comment model and automating the process for models ([admin.py][3]) that will be added in the future

Adding commenting model part and model registration in the admin.py file

**for reviewing PR through the admin interface**:
1. vagrant up 
2. vagrant ssh 
3. cd \vagrant\
4. pipenv run python manage.py createsuperuser
5. type details (name, email and a password)
6. go to localhost:8000/admin (it's the same as our usual admin page)
7. add a comment by pressing the '+' button next to the Comments table button 
7. add a post by clicking on the '+' button on the comment table (it will contain only id number for now, and click the save button) and click save on the window that opens
8. fill the required fields (of the comment - the active field is for later for approving comments in the views.py)
9. click the save button, and look for the created comment object on the admin interface (click on the comments category to do so
![example](https://user-images.githubusercontent.com/48455448/113622863-341bcd80-9666-11eb-86fc-6974097e8150.PNG)
)
* for now the other default tabled that has been generated are functioning partially, if someone knows something about it, please share us in the discussion below

**Also regarding the 3 hours difference is time,  it's not a problem- django saves the UTC and the converts it based on the timezone, Israel 
 is UTC +3**
**instructions file to test the new test file for migrations**
[testing-migrations- instructions.txt](https://github.com/beyond-io/picATrip/files/6271386/testing-migrations-.instructions.txt)

related issue: #44
closes #46

Leeadar commented:
When I click on 'Content types' in the Django administration, I get this error:

![image](https://user-images.githubusercontent.com/72739568/113572159-64407d80-9620-11eb-9e46-31608011c127.png)

@EdDev @Yarboa @guy9050 Do you know what is the reason for that?

Also @maayanhd I think you could add an image of the expected comments table

maayanhd commented:
@Leeadar I fixed the comment adding functionality that has happened lately and attached a photo and instructions on interacting with the Admin interface, still haven't found a solution to the contentTypes table rendering problem, @Yarboa, @EdDev, @guy9050 , is it even necessary for the project, its one of the table that auto-registered by the code I added in the admin.py file. (also the Post table for now is not working- but that's ok because it will be deleted once the Post model will be ready and will be replaced by an import as I explained above)

Yarboa reviewed (commented):
Please add all the commands 1 to 4 into readme into:
bootstrap.sh, no manual commands are needed

maayanhd commented:
@Yarboa I asked Barak about adding such a file of tests and he said we don't need to have this, but since you asked what did you mean other than including in this file actions to populate the database as appeared in a way in the challenge?

Yarboa commented:
> @Yarboa I asked Barak about adding such a file of tests and he said we don't need to have this,

I was not aware to that.
> but since you asked what did you mean other than including in this file actions to populate the database as appeared in a way in the challenge?
Please refer this one Beyond - Tool introduction challenge.pdf
see TASK #5


> what do you my by using model methods, simply writing methods to help populate the database?
In case you need a class method in your model, if needed.
but could be overridden see the sample above.

It is just a practice upload, code with minimal verification.
In case you think it is too complicated please reply with new Issue

Leeadar reviewed (commented):
![image](https://user-images.githubusercontent.com/72739568/113844601-549a7380-979d-11eb-8061-67919759549d.png)

I think it works good!
Just a question,
What do you mean by 'active' ? 
And I also don't sure what the purpose of "Quite" in the tags.

maayanhd commented:
@Leeadar I explained it shortly but for now you'll just have to wait until I will commit changes on the view, it's going to help me after the post model will be working, as part of the functionality of adding a comment related to a specific post, to create a comment without saving it in the database and marking it as not active, and only then commenting it to a post and mar it as active, to make sure each comment is saved to the DB after it has been linked to a post, I used a guide for this part..

Leeadar commented:
> The quite part was suppose to describe the post as a pot of a quite place, I can add more words to describe it if you think it'll be needed or in later version we can add an image that describe each tag for example

So I think you meant 'Quiet' and not 'Quite'.
You can also used the word 'crowded' but I think it's a little bit problematic because it could change in high frequency

maayanhd commented:
yeah, @Leeadar you're right obviously, I'll fix it next force push with the test file of the migration, my bad. should I add it as a "Quiet Place" along the way ?

Shoval-Ovadia commented:
works well. 
* need to rebase!

guy9050 commented on pickATrip_django_apps/admin.py:
```diff
@@ -1,3 +1,32 @@
+        super(ListAdminMixin, self).__init__(model, admin_site)
+
+
+# Register all other models automatically - should stay last in file.
+models = apps.get_models()
+for model in models:
+    admin_class = type('AdminClass', (ListAdminMixin, admin.ModelAdmin), {})
+    try:
+        admin.site.register(model, admin_class)
+    except admin.sites.AlreadyRegistered:
+        pass
```
what is this for?

maayanhd replied:
for automating the model registration in the admin for models that will be added later, in case they weren't registered manually as I did with more customization for the comment model, It only provides a way to interact with the models and viewing each table by its fields in the admin interface

Yarboa replied:
There is a way to do that, not needed to be admin
Please refer Task#5 in first challenge

Yarboa replied:
Sorry, can you elaborate here? it is not clear to me

Yarboa replied:
OK, reading this https://docs.djangoproject.com/en/3.1/ref/contrib/admin/
I think it should be removed from this PR, and proposed on different PR,
Can you open an Issue? such as, create  automatic admin interface w/ PseudoCode?
Please implement one thing in single PR

Yarboa replied:
@maayanhd @guy9050, any updates here? 
I choose not die on that hill :+1:

Yarboa commented:
> Please see comment below, no manual commands are needed in PR description
> 
> Consider adding some content into the tables, through
> I would consider this as must adding 0002_test_data.py file which populates the model
> and use some model methods.

I see this, Ack thanks

Yarboa commented:
Sorry, my mistake, there are merge conflicts, can you resolve?

Yarboa commented:
> @Yarboa I asked Barak about adding such a file of tests and he said we don't need to have this, but since you asked what did you mean other than including in this file actions to populate the database as appeared in a way in the challenge?
> what do you my by using model methods, simply writing methods to help populate the database?

I see this now, thanks 0002_test_data.py

maayanhd commented:
@Yarboa I'll resolve conflicts and separate the code you said to a different PR, the idea was that I will handle the registration model process, so I thought it would be not enough code to hold a separate PR.
There's only one thing left, unfortunately I would need to wait to PR #56 since the empty post class would be replaced with an import if the soon to be merged Post model, what do you think? Is there any other way you suggest on doing?

Yarboa commented:
> @Yarboa I'll resolve conflicts and separate the code you said to a different PR, the idea was that I will handle the registration model process, so I thought it would be not enough code to hold a separate PR.

Please rebase, remove the code.
Open an Issue.

> There's only one thing left, unfortunately I would need to wait to PR #56 since the empty post class would be replaced with an import if the soon to be merged Post model, what do you think? Is there any other way you suggest on doing?

Your PR is close to be merged. 
on the Post PR, it will be fixed, ok?

Yarboa reviewed (approved):
Please add all the commands in the readme into:
bootstrap.sh, no manual commands are needed

maayanhd commented:
@Yarboa working on it, had some issues with rebasing and squashing a commit with an older one, I believe soon I' separate the PR as needed, but what about the commands? are you talking about my instructions file? it was same as Barak presented us in the challenge since we didn't had the admin interface, I just added it in case someone would want to review this PR using this commands manually, is it really needed in the PR?

maayanhd commented:
I'm done with the changes after rebasing and all of the stuff, just tell me @Yarboa if adding the command to bootstrap file is needed since, it was only for interacting with the shell and the test file is doing a better job at populating the database as you asked

Yarboa commented:
> I'm done with the changes after rebasing and all of the stuff, just tell me @Yarboa if adding the command to bootstrap file is needed since, it was only for interacting with the shell and the test file is doing a better job at populating the database as you asked me to do

Ack, no need, Good job,
